app/javascript/src/components/Dashboard/Notes/NotesList.jsx

```diff
@@ -1,35 +1,103 @@
-import React, { useState } from "react";
+import React from "react";
+
+import EmptyNotesListImage from "images/EmptyNotesList";
+
+import EmptyState from "components/Common/EmptyState";
 
 import Note from "./Note";
-import EditNotePane from "./Pane/Edit";
 
-const NotesList = ({ notes, user, fetchNotes }) => {
-  const [showEditNote, setShowEditNote] = useState(false);
-  const [selectedNote, setSelectedNote] = useState({});
+const DUMMY_TAGS = [
+  "Gettings Started",
+  "Onboarding",
+  "User Flow",
+  "UX",
+  "Bugs",
+  "V2",
+];
+const STATUS = ["Drafted", "Created", "Updated"];
+const NAMES = [
+  "Debra",
+  "Allen",
+  "Gerald",
+  "Harris",
+  "Raymond",
+  "Carter",
+  "Jacqueline",
+  "Torres",
+];
 
-  const onEditClickHandler = selectedNote => {
-    setSelectedNote(selectedNote);
-    setShowEditNote(true);
+const NotesList = ({
+  notes,
+  onEditClick,
+  onDeleteClick,
+  setShowNewNotePane,
+}) => {
+  // Only for data creation purpose
+  const chooseRandomTags = () => {
+    const shuffled = DUMMY_TAGS.sort(() => 0.5 - Math.random());
+    const randomNumber = Math.floor(
+      Math.random() * (DUMMY_TAGS.length - 1) + 1
+    );
+
+    return shuffled.slice(0, randomNumber);
   };
 
+  // Only for data creation purpose
+  const addTagsToNote = allNotes =>
+    allNotes.map(note => {
+      const tags = chooseRandomTags();
+      return { ...note, tags };
+    });
+
+  // Only for data creation purpose
+  const addStatusToNote = allNotes =>
+    allNotes.map(note => {
+      const randomNumber = Math.floor(Math.random() * STATUS.length);
+      return { ...note, status: STATUS[randomNumber] };
+    });
+
+  // Only for data creation purpose
+  const addUserToNote = allNotes =>
+    allNotes.map(note => {
+      const randomNumber1 = Math.floor(Math.random() * NAMES.length);
+      const randomNumber2 = Math.floor(Math.random() * NAMES.length);
+      return {
+        ...note,
+        user: {
+          firstName: NAMES[randomNumber1],
+          lastName: NAMES[randomNumber2],
+        },
+      };
+    });
+
+  if (notes.length) {
+    notes = addTagsToNote(notes);
+    notes = addStatusToNote(notes);
+    notes = addUserToNote(notes);
```

Comment:
This a very serious mistake. You are directly mutating the state over here by just assigning. This defeats the purpose of make `notes` a state in React!